File: Codes/3075-maximize-happiness-of-selected-children.go
/*
 * @Author: LetMeFly
 * @Date: 2025-12-25 12:56:03
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-12-25 13:08:10
 */
package main

import "sort"

func maximumHappinessSum(happiness []int, k int) (ans int64) {
    sort.Slice(happiness, func(i, j int) bool { return happiness[i] > happiness[j] })
    for i := 0; i < k; i++ {
        happiness[i] -= i
        if happiness[i] <= 0 {
            return
        }
        ans += int64(happiness[i])
    }
    return
}

File: Codes/3075-maximize-happiness-of-selected-children.cpp
/*
 * @Author: LetMeFly
 * @Date: 2025-12-25 12:56:03
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-12-25 12:59:03
 */
#if defined(_WIN32) || defined(__APPLE__)
#include "_[1,2]toVector.h"
#endif

typedef long long ll;
class Solution {
public:
    ll maximumHappinessSum(vector<int>& happiness, int k) {
        sort(happiness.begin(), happiness.end(), greater<>());
        ll ans = 0;
        for (int i = 0; i < k; i++) {
            ans += max(0, happiness[i] - i);
        }
        return ans;
    }
};